fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java

```diff
@@ -938,6 +938,38 @@ private void analyzeAddPartition(OlapTable olapTable, List<PartitionDesc> partit
         }
     }
 
+    private int calAvgBucketNumOfRecentPartitions(OlapTable olapTable, int recentPartitionNum) {
+        int avgBucketNum = 0;
+        if (olapTable.getPartitions().size() < 5) {
+            int backendNum = GlobalStateMgr.getCurrentSystemInfo().getBackendIds().size();
+            // When POC, the backends is not greater than three most of the time.
+            // The bucketNum will be given a small multiplier factor for small backends.
+            if (backendNum <= recentPartitionNum) {
+                avgBucketNum = 2 * backendNum;
+            } else if (backendNum <= 6) {
+                avgBucketNum = (int) (1.5 * backendNum);
+            } else if (backendNum <= 12) {
+                avgBucketNum = (int) (1.2 * backendNum);
+            } else {
+                avgBucketNum = backendNum;
+            }
+            return avgBucketNum;
+        }
+
+        List<Partition> partitions = (List<Partition>) olapTable.getRecentPartitions(recentPartitionNum);
```

Comment:
If users create 100 partitions first，the 6-100 partition will have only 1 buckets because all the partitions are empty at the begining.